quartz/styles/base.scss

```diff
@@ -261,7 +261,7 @@ thead {
 
   article > & > a {
     color: var(--dark);
-    &.internal {
+    &.internal:not([class~="alias"]) {
```

Comment:
oh this also happens to the text as well. Do you think we should also add a class to all of internal alias as well?

Reply:
i'm not sure what you mean? `internal` is a class already and all `internal` should have that background styling (with the exception of the anchor link special case)

Reply:
also I guess not every case is an alias, there might be just internal link here.

This is probably not a great solution either.

Reply:
i meant that we should probably just edit https://github.com/jackyzha0/quartz/blob/34334eabed151cf5fa881d6b34f75107a2d3871e/quartz/plugins/transformers/gfm.ts#L34 directly to add the class here and just remove background on this (this is the only case i think we dont want background)

Reply:
ah i see. will do.

Reply:
hmm this solve a different problem though, but I will add this to the other PR.

Reply:
Wait actually I don't follow.

```md
## note -- [[path/note|LLMs]]
```

The class would be handled in ofm.ts first no?

so we have to inject the class there?

Reply:
no we just change the selector in L264 of `base.scss` to `&.anchor` because that's the only class we don't want a background

by making that selector more specific, any other links in the header should have the appropriate bg

Reply:
Correct me if i'm wrong, but the headers we don't need any background.

But we still want background for any internal link.

We will set the class "anchor" for all internal links that is within the parent node (which in this case is the header div)

And you suggest to add the class `.anchor` to gfm.ts

afaict adding classes into L34 won't work, because that is the class of the icon, not the item within the header.

Reply:
or are you suggesting that we add another entry into gfm to handle this case?

because we need access to the parent node itself to do this right?

Reply:
Actually I think I understand what you meant here.

Reply:
PTAL :)

Reply:
![image](https://github.com/jackyzha0/quartz/assets/23178940/f7ad7f06-e412-45a4-a102-37809ae357d6)

you need to add the class on the link icon too (also let's fix the link colour too)

Reply:
basically my desired state is that links in headers are styled pretty much like normal links and the header anchor icon is styled as it is today

Reply:
can you fetch again? I think i fix that

Reply:
just pulled, same result